Zend/zend_compile.h

```diff
@@ -173,7 +173,14 @@ typedef struct _zend_try_catch_element {
 	uint32_t finally_end;
 } zend_try_catch_element;
 
+#define ZEND_LIVE_TMPVAR  0
+#define ZEND_LIVE_LOOP    1
+#define ZEND_LIVE_SILENCE 2
+#define ZEND_LIVE_ROPE    3
+#define ZEND_LIVE_MASK    3
```

Comment:
MASK is used to extract low two bits. they may represent one of four values
(TMPVAR, LOOP, SILENCE and ROPE).
This is a bit stealing technique.

On Thu, Nov 12, 2015 at 10:13 PM, Markus Staab <notifications@github.com>
wrote:

> In Zend/zend_compile.h
> <https://github.com/php/php-src/pull/1634#discussion_r44700420>:
>
> > @@ -173,7 +173,14 @@ typedef struct _zend_try_catch_element {
> >  	uint32_t finally_end;
> >  } zend_try_catch_element;
> >
> > +#define ZEND_LIVE_TMPVAR  0
> > +#define ZEND_LIVE_LOOP    1
> > +#define ZEND_LIVE_SILENCE 2
> > +#define ZEND_LIVE_ROPE    3
> > +#define ZEND_LIVE_MASK    3
>
> Intentionally the same value for ROPE & MASK?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/1634/files#r44700420>.
>